### DISCUSSION TITLE
Student wanting to help by contributing a small amount.

### DISCUSSION THREAD

**Discussion by falzfalzfalz** [pocketbase/pocketbase#1021] [Q&A] [answered]
Hello there,

I'm a **student** currently doing my master thesis in cybersec. For one of my course we have to **contribute to an open-source project of our choice**. The goal is to have an approved pull request by the end of November. I'm really really interested in PocketBase for lots of reasons, and I'm using it on the daily in my side projects. Therefore, **I would like to know if there's some small tasks (writing documentation, correcting some UI/UX bugs, ...) that need work and that I could help with**. This is my first time contributing to such a project, I'm sorry if it is not the way I should go about it.

I have a fair share of experience with **Svelte** and **SvelteKit** but not a lot with Go, I thinks it's useful to know.

Have a good day,

Oscar

**Comment by benallfree** [comment]
Hi @shaparder, if you don’t find anything with PocketBase directly, please consider https://pockethost.io. We are building a hosting platform for PocketBase and have lots of Svelte development coming in the next few weeks.

**Reply by falzfalzfalz** [reply]
Hi @benallfree, thanks for letting me know ! I might be interested in helping, as I also have experience in docker and containers.

What is the goal of pockethost ? Is it to offer the same hosting and deployment services as supabase but with pocketbase ? Do you plan on making it a paid service at the end of the road ? If so, is it okay with @ganigeorgiev ?

Thanks for your answer !

**Reply by ganigeorgiev** [reply]
> If so, is it okay with @ganigeorgiev?

Yes, this is completely fine by me. The project is licensed under MIT and everyone is free to offer it as a service if they want to. I've received several similar questions on the support mail, but I think @benallfree's pockethost.io is the only open-source one so far.

**Reply by benallfree** [reply]
@shaparder Yes the goal is to offer a hosted experience for PocketBase. Although I like the sovereignty of self-hosting, there is an undeniable benefit to a managed service that handles all the backend considerations. As @ganigeorgiev said the project is open source and anyone can spin up a hosting service if they want to. I'm not aware of anyone using my code that way, but they are welcome to do so. Paid support is planned in the roadmap, but for now I'm just focusing on the free tier and making everything run smoothly.

There are currently about 130 users and 200 instances running on pockethost. Of those, our new stats collection tells us that about 25 of those instances (~15%) are heavily active. pockethost is beginning to show viability as a developer playground and possible production environment.

In any case, it sounds like you might contribute some documentation to the core PocketBase project. That's awesome!

**Reply by falzfalzfalz** [reply]
> @shaparder Yes the goal is to offer a hosted experience for PocketBase. Although I like the sovereignty of self-hosting, there is an undeniable benefit to a managed service that handles all the backend considerations. As @ganigeorgiev said the project is open source and anyone can spin up a hosting service if they want to. I'm not aware of anyone using my code that way, but they are welcome to do so. Paid support is planned in the roadmap, but for now I'm just focusing on the free tier and making everything run smoothly.

I totally agree with that ! There's a gap that is getting filled with this project, it's great.

> There are currently about 130 users and 200 instances running on pockethost. Of those, our new stats collection tells us that about 25 of those instances (~15%) are heavily active. pockethost is beginning to show viability as a developer playground and possible production environment.

Congratulations on getting it up and running, and also on getting the user base already this big, that's impressive !

> In any case, it sounds like you might contribute some documentation to the core PocketBase project. That's awesome!

I would definitely be interested in working on PocketHost too, let's talk about it on the PocketHost repo !

**Comment by ganigeorgiev** [comment, marked as answer]
Hi, there are a lot of things that could be improved in PocketBase (you could check the [v1.0.0 roadmap](https://github.com/orgs/pocketbase/projects/2)) but they are all mostly Go related and require some additional research and discussion before even begin implementing them and therefore it may not be suitable considering your deadline.
General documentation help, eg. fixing typos and suggesting better explanations, will be needed but you may want to wait for the new docs related to the upcoming v0.8.0 release (if nothing serious shows up, most likely this weekend).

**Reply by falzfalzfalz** [reply]
Thanks for your answer, I will probably look for opportunities to help on the documentation while learning Go.
